pkg/sql/delegate/show_ranges.go

```diff
@@ -287,6 +287,12 @@ AND s.end_key > r.start_key`)
 			fmt.Fprintf(&buf, " AND s.index_id = %d", idx.ID())
 		}
 	}
+
+	// Exclude dropped tables from .crdb_internal.table_spans
+	if n.Source != tree.ShowRangesIndex && n.Options.Mode != tree.ExpandIndexes {
+		buf.WriteString(" AND s.dropped = false")
```

Comment:
I'm not a SQL expert, but I learned when you put WHERE instead of AND it forced the resulting query to a INNER JOIN.